server/src-exec/Main.hs

```diff
@@ -101,48 +102,88 @@ printJSON = BLC.putStrLn . A.encode
 printYaml :: (A.ToJSON a) => a -> IO ()
 printYaml = BC.putStrLn . Y.encode
 
-mkAuthMode :: Maybe AccessKey -> Maybe Webhook -> Maybe T.Text -> Either String AuthMode
-mkAuthMode mAccessKey mWebHook mJwtSecret =
+mkAuthMode
+  :: ( MonadIO m
+     , MonadError T.Text m
+     )
+  => Maybe AccessKey
+  -> Maybe Webhook
+  -> Maybe T.Text
+  -> HTTP.Manager
+  -> LoggerCtx
+  -> m AuthMode
+mkAuthMode mAccessKey mWebHook mJwtSecret httpManager lCtx =
   case (mAccessKey, mWebHook, mJwtSecret) of
-    (Nothing,  Nothing,   Nothing)     -> return AMNoAuth
-    (Just key, Nothing,   Nothing)     -> return $ AMAccessKey key
-    (Just key, Just hook, Nothing)     -> return $ AMAccessKeyAndHook key hook
-    (Just key, Nothing,   Just jwtConf) -> do
-      -- the JWT Conf as JSON string; try to parse it
-      config <- A.eitherDecodeStrict $ TE.encodeUtf8 jwtConf
-      return $ AMAccessKeyAndJWT key config
-
+    (Nothing,  Nothing,   Nothing)      -> return AMNoAuth
+    (Just key, Nothing,   Nothing)      -> return $ AMAccessKey key
+    (Just key, Just hook, Nothing)      -> return $ AMAccessKeyAndHook key hook
+    (Just key, Nothing,   Just jwtConf) -> mkJwtMode key jwtConf httpManager lCtx
     (Nothing, Just _, Nothing)     -> throwError $
       "Fatal Error : --auth-hook (HASURA_GRAPHQL_AUTH_HOOK)"
-      ++ " requires --access-key (HASURA_GRAPHQL_ACCESS_KEY) to be set"
+      <> " requires --access-key (HASURA_GRAPHQL_ACCESS_KEY) to be set"
     (Nothing, Nothing, Just _)     -> throwError $
       "Fatal Error : --jwt-secret (HASURA_GRAPHQL_JWT_SECRET)"
-      ++ " requires --access-key (HASURA_GRAPHQL_ACCESS_KEY) to be set"
+      <> " requires --access-key (HASURA_GRAPHQL_ACCESS_KEY) to be set"
     (Nothing, Just _, Just _)     -> throwError
       "Fatal Error: Both webhook and JWT mode cannot be enabled at the same time"
     (Just _, Just _, Just _)     -> throwError
       "Fatal Error: Both webhook and JWT mode cannot be enabled at the same time"
 
+-- is this correct ?
+mkJwtMode
+  :: ( MonadIO m
+     , MonadError T.Text m
+     )
+  => AccessKey
```

Comment:
Remove the access key argument to this function. It is only used to construct AuthMode. This function can instead return JWTCtx without taking in accesskey.